### ISSUE TITLE
Allow dynamic log path location in `.bazelrc`

### ISSUE THREAD

**Issue by ssarad** [bazelbuild/bazel#22267]
### Description of the feature request:

Allow `.bazelrc` file to have a variable that points to the root of the project (for example: `%workspace`). Unfortunately `%workspace%` does not work when specifying the path to logs as it appends %workspace% as a string, not a variable. 

```sh
build:trace --execution_log_json_file=%workspace%/Bazel/tmp/logs/execution.json
```

<hr>

A currently work around is to edit `tools/bazel`, grab the current path to root of the project and echo the following to a `Bazel/tmp/local.bazelrc` file 

```sh
echo "build:trace --execution_log_json_file=$script_root/../Bazel/tmp/logs/execution.json" > Bazel/tmp/local.bazelrc
```
Then, import `local.bazelrc`in the project's `.bazelrc` using `try-import`

```sh
try-import %workspace%/Bazel/tmp/xcode.bazelrc
```

### Which category does this issue belong to?

CLI, Configurability

### What underlying problem are you trying to solve with this feature?

Currently, if we want to specify a location to the logs, we can add the following in the `.bazelrc` file

```sh
build:trace --execution_log_json_file=Bazel/tmp/logs/execution_logs.json
```
This works fine if a Bazel command is executed from the root of the project but if we run a Bazel command from a subdirectory inside the project, Bazel attempts to search for the path in such manner:

```sh
pwd + Bazel/tmp/logs/execution_logs.json
```

This makes the path mention to dump the logs in the `.bazelrc` file as an invalid path

### Which operating system are you running Bazel on?

MacOS Sonoma 14.4.1

### What is the output of `bazel info release`?

INFO: Invocation ID: c18d31dc-3cbd-4cc0-864b-aff070099dee release 6.4.0

**Comment by keith** [comment]
This issue is worse for `experimental_execution_log_compact_file`, where a relative path to a directory causes a build failure instead of it just silently not writing the file:

```
ERROR: /path/to/repo/nested/dir/logs/execution.log (No such file or directory)
```
in this example what I want is the log to be produced in `/path/to/repo/logs/execution.log`, but when im working in the nested directory (`cd nested/dir`) it tries to create that relatively, and the `logs` dir doesn't exist.

**Comment by keith** [comment]
fix here https://github.com/bazelbuild/bazel/pull/23813

**Comment by ssarad** [comment]
You the man @keith ! Thanks for the PR 🫡

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
